File: Docs/Reference/Comparisons/fib.m
#import <Foundation/Foundation.h>

int main(int argc, char *argv[]) {
    __block double (^fib)(double n);
    fib = ^(double n) {
        return n <= 1 ? n : fib(n - 1.0) + fib(n - 2.0);
    };
    fib(35);
    return 0;
}

File: Docs/Reference/Comparisons/fib.rb
def fib(n, a=0, b=1)
#    if n == 0 then
#        return a
#    else
#        return fib(n-1, b, a+b)
#    end
    if n < 2 then
        return n
    else
        return fib(n-1) + fib(n-2)
    end
end
p fib(35)

#fib = lambda { |n|
#    if n < 2 then
#        return n
#    else
#        return fib[n-1] + fib[n-2]
#    end
#}
#p fib[35]

